src/snakemake/dag.py

```diff
@@ -960,9 +961,17 @@ def unneeded_files():
 
         for f in unneeded_files():
             if self.workflow.dryrun:
-                logger.info(f"Would remove temporary output {fmt_iofile(f)}")
+                if (
+                    not self.workflow.output_settings.quiet
+                    or Quietness.RULES not in self.workflow.output_settings.quiet
+                ):
+                    logger.info(f"Would remove temporary output {fmt_iofile(f)}")
             else:
-                logger.info(f"Removing temporary output {fmt_iofile(f)}.")
+                if (
+                    not self.workflow.output_settings.quiet
+                    or Quietness.RULES not in self.workflow.output_settings.quiet
+                ):
+                    logger.info(f"Removing temporary output {fmt_iofile(f)}.")
```

Comment:
I don't think quietness should be handled explicitly outside of the logger code, rather via setting the log event type. Do you agree @cademirch?

Reply:
I went ahead and committed the quietness solution I proposed above. I think this is the best way to handle quietness, since its the most explicit, and does not depend on LogEvent (which is mainly for informing formatting and context). If you agree with this approach @johanneskoester, lets merge this and then I can tackle updating all the other logger calls in #3571 or another PR.

Reply:
Great idea!